### PR TITLE
xtensa build fixes

### PR DESCRIPTION
**Pull request by finikorg**
Set of fixes from xtensa ADSP

### CODE REVIEW

**Comment by andrewboie** [comment]
hmmm, this seems to be breaking some posix tests.

**Review by jhedberg — APPROVED** [review]

**Review by pfalcon — CHANGES_REQUESTED** [review]
> xtensa build fixes
> Set of fixes from xtensa ADSP

Thanks for the patch. What is ADSP and how it relates to "build fixes"? And what issues this PR actually fixes? Please include description of those both in the PR description, to help with reviewing, and in the commit messages of relevant commits, to help with investigation of any issues in the future.

A specific commit:

> libc: minimal: Use zephyr include instead of system include

> Using zephyr_include_directories() instead of
zephyr_system_include_directories places include before toolchain
include for some setup.

This doesn't seem to be related to Xtensa, yet it's included in "Xtensa fixes" PR. Did you analyze how it may affect other targets (ping to @SebastianBoe is in that regard)? Again, no explanation of what exact issue it fixes. And just as a complete hypothesizing, could it be that the commit message contains a good hint of the actual issue, that "**some setup**"" has issues, and that setup is what needs fixing, instead of applying global changes?

**Comment by pfalcon** [comment]
To elaborate on my concerns, here's what my working copy contains in `zephyr/tests/posix/common/`:

~~~
-rw-rw-r-- 1 pfalcon pfalcon 189750 Oct  8  2018 clock.c.out
-rw-rw-r-- 1 pfalcon pfalcon 189186 Oct  8  2018 clock.c.out.bad
-rw-rw-r-- 1 pfalcon pfalcon    246 Oct 15 14:24 CMakeLists.txt
-rw-rw-r-- 1 pfalcon pfalcon   1951 Jun  6 20:56 cmd.2
-rw-rw-r-- 1 pfalcon pfalcon   1930 Jun  6 16:16 cmd-w
-rw-r--r-- 1 pfalcon pfalcon   1936 Jun  6 16:13 cmd-wo
-rw-rw-r-- 1 pfalcon pfalcon   1942 Jun  6 21:04 log
drwxrwxr-x 3 pfalcon pfalcon   4096 Aug 27 15:45 outdir
~~~

clock.c.out vs clock.c.out.bad are C processor output where I debugged why one build is ok and another gets bad definitions (bad headers picked up, wrong include order, etc.). cmd.2, cmd-w, cmd-wo, log are being doing the same debugging, on the level of 2K-sized compile command lines. All current POSIX include paths are result of careful incremental adjustment, juggling to keep multiple pieces from falling. So, please forgive concerns that not carefully planned changes may break something again. Detailed descriptions of the problems being fixed, and logic why exactly a specific change was applies would help with future investigations. And I'd still be prepared that changes may need to be reverted if found that they regress other things.

**Comment by pfalcon** [comment]
7d2df83b5184b4566d43dd8a8d9abeb7c1d85694 . So you're effectively reverting change done previously. This can't be right. If the Xtensa is the only target which has toolchain includes in the order inconsistent wrt to the other toolchains, that's where problem is.

Overall, we have at least 3 categories of includes:

1. Toolchain includes.
2. System includes (for stuff like libc).
3. Application-level includes (for everything else).

All these 3 categories are needed and should be handled/managed properly. We can't shove system includes under application ones, and expect things to work. Likewise, toolchain headers should go before system headers in the search path.

File: lib/libc/minimal/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_system_include_directories(include)
```
[the remark this answers is not in the document]
Comment (pfalcon) [reply lib/libc/minimal/CMakeLists.txt]: Nevermind, it's introduced in https://github.com/zephyrproject-rtos/zephyr/commit/7d2df83b5184b4566d43dd8a8d9abeb7c1d85694 by myself, explaining why it's needed. Further discussion is at https://github.com/zephyrproject-rtos/zephyr/pull/20050#issuecomment-546239184

**Comment by finikorg** [comment]
I will check can this be done other way

**Pull request closed without merging**